diff --git a/e2e/quantized_llama/modeling_llama.py b/e2e/quantized_llama/modeling_llama.py
index 3e83ef0..a592ef4 100644
--- a/e2e/quantized_llama/modeling_llama.py
+++ b/e2e/quantized_llama/modeling_llama.py
@@ -52,7 +52,7 @@ class QuarotFP16LlamaAttention(LlamaFlashAttention2):
 
         kv_seq_len = key_states.shape[1]
         kv_seq_len += past_key_value.get_usable_length(kv_seq_len, self.layer_idx)
-        cos, sin = self.rotary_emb(value_states, seq_len=kv_seq_len)
+        cos, sin = self.rotary_emb(value_states, position_ids, seq_len=kv_seq_len)
         query_states, key_states = apply_rotary_pos_emb(query_states, key_states, cos, sin, position_ids, unsqueeze_dim=2)
 
         past_key_value = getattr(self, "past_key_value", past_key_value)
