diff --git a/setup.py b/setup.py
index 233f95b..bc483fb 100644
--- a/setup.py
+++ b/setup.py
@@ -23,6 +23,7 @@ def get_cuda_arch_flags():
         '-gencode', 'arch=compute_75,code=sm_75',  # Turing
         '-gencode', 'arch=compute_80,code=sm_80',  # Ampere
         '-gencode', 'arch=compute_86,code=sm_86',  # Ampere
+        '-gencode', 'arch=compute_89,code=sm_89',  # Ada
     ]
     
 def third_party_cmake():
@@ -34,7 +35,7 @@ def third_party_cmake():
     if cmake is None:
             raise RuntimeError('Cannot find CMake executable.')
 
-    retcode = subprocess.call([cmake, HERE])
+    retcode = subprocess.call([cmake, '-DCMAKE_CUDA_ARCHITECTURES=89', HERE])
     if retcode != 0:
         sys.stderr.write("Error: CMake configuration failed.\n")
         sys.exit(1)
